### PR TITLE
Add stack support

### PR DESCRIPTION
**Pull request by turion**

### CODE REVIEW

File: travis-build.sh [hunk 1]
```diff
@@ -1,15 +1,19 @@
...
-fi
+# Cabal
+
+cabal new-build all --enable-tests
+cabal new-test all --enable-tests
+# git fetch origin master:origin/master && git rebase origin/master --exec "cabal new-build all --enable-tests"
+
+# Stack
+
+export STACK_YAML="stack.${TRAVIS_HASKELL_VERSION}.yaml"
```
Comment (miguel-negrao) [inline travis-build.sh:11]: this can be set directly using `stack build --system-ghc --test --stack-yaml "stack.${TRAVIS_HASKELL_VERSION}.yaml"`.

Comment (turion) [reply travis-build.sh:11]: Good to know! But I'll leave it like this since I'll be using the setting twice.

File: travis-build.sh [hunk 2]
```diff
@@ -0,0 +1,19 @@
+#!/bin/bash
+
+# Cabal
+
+cabal new-build all --enable-tests
+cabal new-test all --enable-tests
+# git fetch origin master:origin/master && git rebase origin/master --exec "cabal new-build all --enable-tests"
```
Comment (miguel-negrao) [inline travis-build.sh:7]: I think that given that now this PR is targeting master, this can be uncommented, and it should build correctly again.

**Comment by miguel-negrao** [comment]
Should the older individual stack files be removed ?

```
find . -name 'stack*yaml'
./demos/stack.yaml
./essence-of-live-coding-gloss-example/stack.yaml
./essence-of-live-coding-gloss/stack.yaml
./essence-of-live-coding-pulse-example/stack.yaml
./essence-of-live-coding-pulse/stack.yaml
./essence-of-live-coding-quickcheck/stack.yaml
./essence-of-live-coding-speedtest-yampa/stack.yaml
./gears/stack.yaml
./stack.8.10.3.yaml
./stack.8.6.5.yaml
./stack.8.8.4.yaml
./stack.yaml
```

**Comment by turion** [comment]
> Should the older individual stack files be removed ?
> 
> ```
> find . -name 'stack*yaml'
> ./demos/stack.yaml
> ./essence-of-live-coding-gloss-example/stack.yaml
> ./essence-of-live-coding-gloss/stack.yaml
> ./essence-of-live-coding-pulse-example/stack.yaml
> ./essence-of-live-coding-pulse/stack.yaml
> ./essence-of-live-coding-quickcheck/stack.yaml
> ./essence-of-live-coding-speedtest-yampa/stack.yaml
> ./gears/stack.yaml
> ./stack.8.10.3.yaml
> ./stack.8.6.5.yaml
> ./stack.8.8.4.yaml
> ./stack.yaml
> ```

I guess so. Or is it useful if e.g. some people only want to build the `pulse` backend without spending time to compile gloss, OpenGL etc.?

**Comment by miguel-negrao** [comment]
> I guess so. Or is it useful if e.g. some people only want to build the `pulse` backend without spending time to compile gloss, OpenGL etc.?

That can be done by specifying which target to build:

```
stack build essence-of-live-coding-pulse
```

this will build essence-of-live-coding-pulse and its dependencies, including essence-of-live-coding.

**Comment by miguel-negrao** [comment]
@turion let me know if you want to me to do additional changes in this PR.

**Comment by turion** [comment]
Thanks, I lost track of this.

**Pull request merged**